src/main/java/org/folio/utils/CalculationUtils.java

```diff
@@ -107,10 +109,10 @@ private static Map<String, BigDecimal> calculate(Double dInitialAllocation,
     BigDecimal awaitingPayment = BigDecimal.valueOf(dAwaitingPayment);
 
     BigDecimal allocated = initialAllocation.add(allocationTo).subtract(allocationFrom);
-    BigDecimal unavailable = encumbered.add(awaitingPayment).add(expended);
+    BigDecimal unavailable = ensureNonNegative(encumbered.add(awaitingPayment).add(expended).subtract(credited));
     BigDecimal totalFunding = allocated.add(netTransfers);
-    BigDecimal cashBalance = totalFunding.subtract(expended);
-    BigDecimal available = totalFunding.subtract(unavailable);
+    BigDecimal cashBalance = totalFunding.subtract(expended).add(credited);
+    BigDecimal available = totalFunding.subtract(encumbered.subtract(awaitingPayment).add(expended).subtract(credited));
```

Comment:
this should be:
available = totalFunding.subtract(encumbered.add(awaitingPayment).add(expended).subtract(credited));
(we are using unavailable as before but without the "positive or zero" function)

Reply:
Good catch, fixed